### DISCUSSION TITLE
Repair Minigames and Skill levels.

### DISCUSSION THREAD

**Discussion by mjsonic77** [FakeFishGames/Barotrauma#6594] [Ideas and suggestions]
There should be a mini-game for fixing stuff, based on whether it's electrical or mechanical.

**Examples**

- You have to solder broken wires.
- Drain the water out the junction box.
- Renavigate said wires back to a color coded input.
- You have to fix some lose screws on the fabricator.
- Weld holes in one of the busted pumps.
- Clean out the pumps so they can drain properly.

If you don't have the required skill level you may shake a little when trying to fix a junction box, or you might not weld the hole perfectly shut the first time in the pump.
(These are not definite suggestions of what the mini-games should be.)

**Comment by Saevon** [comment]
The problem is a lot of the repairs are really consant, and doing the same minigame can get very tedious.

Instead I'd propose making the minigame get harder based on: 
- Skill Level:   If you are below the required skill the game gets harder
- Item in question (Reactor, junction, o2, etc)

This could partially replace the existing %failure, tho if its a hard enough task that could still happen

How would it be harder?
- More Punishing: 
- Add time limits to the minigame (Only on skill too low, and difficult machinery)
- More precision
  - wire something wrong it breaks vs wire something wrong and it lets you keep going (X mistakes before it fails)
  - soldering / welding shakes a bit more (low skill)
  - soldering / welding has a smaller "accurate" path to trace
- Slower repair, needs more game played
- Progress saved? or Reset?
  - E.g:   Say you need to fix 6 "wires":  it could update the repairs (⅙ repaired) every "wire", or only if you correctly do 3 (½ repaired), then the other 3.   (So if you'd need 3 successes in a row)


Just needs to be a balance around how much harder something is, versus how much more tedious (and doable) it is.



If oxygen is fleshed out, a repairs task with a "O2 Leak detector"  could be a lot more fun (let you wander the ship)

**Comment by duarm** [comment]
these pointless minigames would become old really fast, at least for me. I would trade all of these minigames and the current one for one in which you need to actually rewire the equipment to fix it. 
They would fail this way only out of combat, and you would need to identify what broke and track down the cause, e.g. The sonar doesnt work, track down the wire with the goggles, find the broken wire, rewire it. Every light turned off: probably the light junction box
